### PR TITLE
big-o-notation-exercise - Jonathan Matsuda

### PR DESCRIPTION
**Pull request by jonathanymatsuda**
Screen Shot 2022-01-10 at 3 40 55 PM

### CODE REVIEW

File: big-o-notation/big-o.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1 = O(1)
+      c < i;                        // 2 * n^2 = O(n^2)
+      c++                           // 2 * n^2 = O(n^2)
+    ) {
+      const comparing = words[c];   // 2 * n^2 = O(n^2)
+      if (comparing === word) {     // 1 * 1 = O(1)
+        isUnique = false;           // 1 * 1 = O(1)
+      }
```
Comment (thebearingedge) [inline big-o-notation/big-o.js:37]: These actually happen up to `n * n` as well, since we're in a nested loop.

File: big-o-notation/big-o.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= 0(n)
+    i++                             // 2 * n = O(2n) ~= 0(n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n) ~= 0(n)
+    if (!seen[word]) {              // 1 * 1 = O(1)
+      seen[word] = true;
+      unique[unique.length] = word; // 3 * n = O(3n)
+    }
```
Comment (thebearingedge) [inline big-o-notation/big-o.js:15]: Everything inside of a loop is going to happen `n` times worst case.

File: big-o-notation/big-o.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= 0(n)
+    i++                             // 2 * n = O(2n) ~= 0(n)
+  ) {
+    const word = words[i];          // 3 * n = O(3n) ~= 0(n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1 = O(1)
```
Comment (thebearingedge) [inline big-o-notation/big-o.js:30]: This is inside of another loop, so it's actually `1 * n`

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by jonathanymatsuda** [comment]
Screen Shot 2022-01-10 at 4 11 52 PM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**